Core.Collectors/IO/RecordWriterCore.cs

```diff
@@ -34,6 +35,9 @@ public abstract class RecordWriterCore<T> : IRecordWriter where T : FunctionCont
         private string currentOutputSuffix;
         private int currentFileIndex;
 
+        // Used to detect threading issues; RecordWriter is not thread-safe
+        private int isActive;
```

Comment:
>isActive [](http://example.com/codeflow?start=20&length=8)

Since this is an 'int' can, you rename this something like writerCount?

Reply:
But it's also not really a count, it's used as a bool with values 0/1. `Interlocked` just doesn't have any overloads for bools.